## Final Project Submission

Please fill out:
* Student name: Peris Kigo
* Student pace: full time
* Scheduled project review date/time:30/06/25
* Instructor name: William Okomba
* Blog post URL:



# Phase_1 Project - Aviation Analysis.
The company wants to expand into the aviation industry to diversify its portfolio. They are specifically interested in purchasing and operating airplanes  , but they are unsure which aircraft are the safest investment.

## Problem statement.
Without the company understanding wwhich aircrafts are most frequently involved in accidents, the company risks making costly and potentially dangerous decisions. To support a safe and successful entry into this industry, there is  aneed to analyze the data given to identify which aircrafts to purchase to help the company make informed and data-driven decisions.
## Goals.
To use the data to help minimize operational risk by selecting aircrafts with the lowest incidents or accidents.
## Project scope.
* Understand what factors contribute to the accidents.
* compare different aircfrafts types based on the accidents.
* Identify aircrafts with the lowest incident rates

# Data Understanding.
The dataset was provided by the National Transportation Safety Board. It contains the aviation accident data from 1962 to 2023 about civil aviation accidents. The data was shared as part of this project and is stored locally in the folder.
The file is provided in a **CSV** format.
> ###      **'Aviation_Data.CSV'**


In [26]:
# importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [27]:
df = pd.read_csv ('AviationData.csv', encoding = 'cp1252', low_memory = False )

In [28]:
df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   
3       EUREKA, CA  United States        NaN         NaN          NaN   
4       Canton, OH  United States        NaN         NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [29]:
df.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
88884  20221227106491           Accident      ERA23LA093  2022-12-26   
88885  20221227106494           Accident      ERA23LA095  2022-12-26   
88886  20221227106497           Accident      WPR23LA075  2022-12-26   
88887  20221227106498           Accident      WPR23LA076  2022-12-26   
88888  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
88884  Annapolis, MD  United States      NaN       NaN          NaN   
88885    Hampton, NH  United States      NaN       NaN          NaN   
88886     Payson, AZ  United States  341525N  1112021W          PAN   
88887     Morgan, UT  United States      NaN       NaN          NaN   
88888     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
88884          NaN  ...          Personal                 NaN   
88885          NaN  ...               NaN                 NaN   
88886       PAYSON  ...          Personal                 NaN   
88887          NaN  ...          Personal  MC CESSNA 210N LLC   
88888          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
88884                  0.0                    1.0                  0.0   
88885                  0.0                    0.0                  0.0   
88886                  0.0                    0.0                  0.0   
88887                  0.0                    0.0                  0.0   
88888                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight Report.Status  \
88884             0.0               NaN                    NaN           NaN   
88885             0.0               NaN                    NaN           NaN   
88886             1.0               VMC                    NaN           NaN   
88887             0.0               NaN                    NaN           NaN   
88888             1.0               NaN                    NaN           NaN   

      Publication.Date  
88884       29-12-2022  
88885              NaN  
88886       27-12-2022  
88887              NaN  
88888       30-12-2022  

[5 rows x 31 columns]

__*Observation*:__ The dataset is uniform from top to bottom

In [30]:
df.shape

(88889, 31)

__*Observation*:__ The dataset comprises of 88,889 records and 31 columns

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

__*Observation*:__ The dataset comprises of 5 numerical columns and 26 categorical columns.

In [32]:
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

__*Observation*__: The injury data is heavily skewed with low or zero value and a few extreme values.

In [33]:
df.describe(include = 'O').T

count unique             top   freq
Event.Id               88889  87951  20001212X19172      3
Investigation.Type     88889      2        Accident  85015
Accident.Number        88889  88863      CEN22LA149      2
Event.Date             88889  14782      1984-06-30     25
Location               88837  27758   ANCHORAGE, AK    434
Country                88663    219   United States  82248
Latitude               34382  25589         332739N     19
Longitude              34373  27154        0112457W     24
Airport.Code           50132  10374            NONE   1488
Airport.Name           52704  24870         Private    240
Injury.Severity        87889    109       Non-Fatal  67357
Aircraft.damage        85695      4     Substantial  64148
Aircraft.Category      32287     15        Airplane  27617
Registration.Number    87507  79104            NONE    344
Make                   88826   8237          Cessna  22227
Model                  88797  12318             152   2367
Amateur.Built          88787      2              No  80312
Engine.Type            81793     12   Reciprocating  69530
FAR.Description        32023     31             091  18221
Schedule               12582      3            NSCH   4474
Purpose.of.flight      82697     26        Personal  49448
Air.carrier            16648  13590           Pilot    258
Weather.Condition      84397      4             VMC  77303
Broad.phase.of.flight  61724     12         Landing  15428
Report.Status          82505  17074  Probable Cause  61754
Publication.Date       75118   2924      25-09-2020  17019

In [34]:
df1 = df.copy(deep = True)
df1[:3]

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   

          Location        Country   Latitude   Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN         NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN         NaN          NaN   
2    Saltville, VA  United States  36.922223  -81.878056          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  

[3 rows x 31 columns]

In [35]:
for coln in df1:
    coln_val = df1[coln].unique()
    print(f'{coln}\n{coln_val}\n')

Event.Id
['20001218X45444' '20001218X45447' '20061025X01555' ... '20221227106497'
 '20221227106498' '20221230106513']

Investigation.Type
['Accident' 'Incident']

Accident.Number
['SEA87LA080' 'LAX94LA336' 'NYC07LA005' ... 'WPR23LA075' 'WPR23LA076'
 'ERA23LA097']

Event.Date
['1948-10-24' '1962-07-19' '1974-08-30' ... '2022-12-22' '2022-12-26'
 '2022-12-29']

Location
['MOOSE CREEK, ID' 'BRIDGEPORT, CA' 'Saltville, VA' ... 'San Manual, AZ'
 'Auburn Hills, MI' 'Brasnorte, ']

Country
['United States' nan 'GULF OF MEXICO' 'Puerto Rico' 'ATLANTIC OCEAN'
 'HIGH ISLAND' 'Bahamas' 'MISSING' 'Pakistan' 'Angola' 'Germany'
 'Korea, Republic Of' 'Martinique' 'American Samoa' 'PACIFIC OCEAN'
 'Canada' 'Bolivia' 'Mexico' 'Dominica' 'Netherlands Antilles' 'Iceland'
 'Greece' 'Guam' 'Australia' 'CARIBBEAN SEA' 'West Indies' 'Japan'
 'Philippines' 'Venezuela' 'Bermuda' 'San Juan Islands' 'Colombia'
 'El Salvador' 'United Kingdom' 'British Virgin Islands' 'Netherlands'
 'Costa Rica' 'Mozambique' 'Jama

__*Observation*:__ The dataset has a lot of nan values and unknown values

In [36]:
df1.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [37]:
df.replace(['UNK' , 'Unknown' , 'Unk' , 'nan' , 'NONE'],np.nan, inplace = True)

In [38]:
df1.isnull().sum()

Event.Id                      0
Investigation.Type            0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Airport.Code              38757
Airport.Name              36185
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Registration.Number        1382
Make                         63
Model                        92
Amateur.Built               102
Number.of.Engines          6084
Engine.Type                7096
FAR.Description           56866
Schedule                  76307
Purpose.of.flight          6192
Air.carrier               72241
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
Report.Status              6384
Publication.Date          13771
dtype: i

__*Observation*:__ Several columns contained inconcistent representation of missing values
* These were standardised by replacing them with actual null values

In [39]:
df[['Make' , 'Model']].isnull().sum()

Make      84
Model    102
dtype: int64

In [40]:
df[df['Make'].isnull() | df['Model'].isnull()]

Event.Id Investigation.Type Accident.Number  Event.Date  \
354    20020917X02286           Accident      LAX82FUA11  1982-02-19   
443    20020917X02032           Accident      FTW82DA082  1982-02-28   
461    20020917X01674           Accident      ATL82DA063  1982-03-02   
835    20020917X02071           Accident      FTW82DA129  1982-04-14   
1637   20020917X03452           Accident      FTW82FPD13  1982-06-20   
...               ...                ...             ...         ...   
73592  20130531X43222           Incident      DCA13WA097  2013-05-22   
73617  20130808X52306           Incident      CEN13WA470  2013-05-26   
74873  20140327X61657           Accident      ERA14LA174  2014-03-26   
77653  20151222X23224           Incident      DCA16WA034  2015-12-04   
77952  20160314X33602           Incident      ENG16WA009  2016-03-10   

                  Location        Country Latitude Longitude Airport.Code  \
354           TORRANCE, CA  United States      NaN       NaN          TOA   
443             TAYLOR, TX  United States      NaN       NaN          T74   
461         BROWNSBORO, AL  United States      NaN       NaN          NaN   
835          CLEVELAND, TX  United States      NaN       NaN          6R3   
1637           ROANOKE, TX  United States      NaN       NaN          NaN   
...                    ...            ...      ...       ...          ...   
73592  Panama City, Panama         Panama      NaN       NaN          NaN   
73617     Geneva, Eswatini       Eswatini      NaN       NaN          NaN   
74873         Patillas, PR  United States  175831N  0066115W          X64   
77653     Perth, Australia      Australia      NaN       NaN          NaN   
77952     Zurich, Eswatini       Eswatini      NaN       NaN          NaN   

              Airport.Name  ... Purpose.of.flight     Air.carrier  \
354               TORRANCE  ...          Personal             NaN   
443       TAYLOR MUNICIPAL  ...          Personal             NaN   
461                    NaN  ...          Personal             NaN   
835    CLEVELAND MUNICIPAL  ...          Personal             NaN   
1637           AERO VALLEY  ...          Personal             NaN   
...                    ...  ...               ...             ...   
73592                  NaN  ...               NaN             NaN   
73617                  NaN  ...               NaN             NaN   
74873     Patillas Airport  ...          Personal  Emerito Guzman   
77653                  NaN  ...               NaN       Scoot Air   
77952                  NaN  ...               NaN             NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
354                    0.0                    0.0                  0.0   
443                    0.0                    0.0                  0.0   
461                    0.0                    0.0                  1.0   
835                    0.0                    0.0                  0.0   
1637                   1.0                    1.0                  0.0   
...                    ...                    ...                  ...   
73592                  0.0                    0.0                  0.0   
73617                  0.0                    0.0                  0.0   
74873                  1.0                    0.0                  0.0   
77653                  0.0                    0.0                  0.0   
77952                  0.0                    0.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
354               1.0               VMC                Landing   
443               1.0               VMC            Maneuvering   
461               0.0               VMC               Approach   
835               1.0               VMC                Takeoff   
1637              0.0               VMC            Maneuvering   
...               ...               ...                    ...   
73592            90.0               NaN    

In [41]:
df.dropna(subset=['Make' , 'Model'], inplace=True)

__*Observation*:__ 141 rows had missingvalues in either make or model columns of which they were dropped 

In [44]:
df1['Make']= df1['Make'].astype(str).str.strip().str.upper()
df1['Model']=df1['Model'].astype(str).str.strip().str.upper()
df1['aircraft_type'] = df1['Make'] + '' + df1['Model']

In [45]:
df1['aircraft_type'].value_counts()

aircraft_type
CESSNA152               2366
CESSNA172               1753
CESSNA172N              1163
PIPERPA-28-140           932
CESSNA150                829
                        ... 
ROE-WAYNEMUSTANG II        1
SPENCERPITTS S-1S          1
WOLFSMITH M/P DSA-1        1
BOEING737-548              1
ROYSE RALPH LGLASAIR       1
Name: count, Length: 18297, dtype: int64

__*Observation*:__ The data had different entries of the Models with different formattings, so it had to be standardised.
* Make and model was grouped in 'aircraft-type'